graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -59,29 +59,27 @@ public String get(@Context HttpHeaders headers) {
             throw new RuntimeException("Unable to read AppConfig template while generating web interface configuration: ", e);
         }
 
-        final String serverTransportUri = serverUriFromHeadersOrDefault(headers);
         final Map<String, Object> model = new HashMap<String, Object>() {{
             put("rootTimeZone", configuration.getRootTimeZone());
-            put("serverUri", serverTransportUri);
+            put("serverUri", buildEndpointUri(headers));
             put("appPathPrefix", "");
         }};
         return engine.transform(template, model);
     }
 
-    private String serverUriFromHeadersOrDefault(HttpHeaders httpHeaders) {
-        final List<String> overrideServerUriHeaders = httpHeaders.getRequestHeader(CK_OVERRIDE_SERVER_URI);
-        if (overrideServerUriHeaders != null && !overrideServerUriHeaders.isEmpty()) {
-            Optional<String> firstHeader = overrideServerUriHeaders.stream().filter(s -> {
+    private String buildEndpointUri(HttpHeaders httpHeaders) {
+        Optional<String> endpointUri = Optional.empty();
+        final List<String> headers = httpHeaders.getRequestHeader(CK_OVERRIDE_SERVER_URI);
+        if (headers != null && !headers.isEmpty()) {
+            endpointUri = headers.stream().filter(s -> {
                 try {
-                    return !Strings.isNullOrEmpty(s) && new URL(s) != null;
-                } catch (MalformedURLException e) {
+                    return new URI(s).isAbsolute();
+                } catch (NullPointerException | URISyntaxException e) {
```

Comment:
I assume you mean the change from `java.net.URL` to `java.net.URI` (and with it the change of the caught exceptions).

The `URL` class doesn't have an equivalent of [`URI#isAbsolute()`](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#isAbsolute--) which checks that the URI contains a scheme component (e. g. `http://` or `https://`). Additionally, this change simplifies the condition in the `try` block.